### PR TITLE
[2.0.0] Small api changes

### PR DESCRIPTION
**Pull request by emartynov**
## Task
Part of the #620 

## Changes
* Remove generics from `Completable.asMaybe` and `Completeable.asObservable`
* Rename parameter `action` to `consumer` (See notes)
* Swap parameters for the observable interval and make default for delay as period value
* Add default parameter values and remove overridden versions

## Notes
* There are more `action` named parameters
* IDE shows me unused function warning - should we suppress it?

### CODE REVIEW

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/maybe/DoOnAfter.kt [hunk 1]
```diff
@@ -73,17 +73,17 @@ fun <T> Maybe<T>.doOnAfterSuccess(action: (T) -> Unit): Maybe<T> =
...
     }
 
 /**
- * Calls the [action] when the [Maybe] signals `onComplete`.
- * The [action] is called **after** the observer is called.
+ * Calls the [consumer] when the [Maybe] signals `onComplete`.
+ * The [consumer] is called **after** the observer is called.
  */
-fun <T> Maybe<T>.doOnAfterComplete(action: () -> Unit): Maybe<T> =
+fun <T> Maybe<T>.doOnAfterComplete(consumer: () -> Unit): Maybe<T> =
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/maybe/DoOnAfter.kt:86]: It is not a consumer because it does not consume anything, so the name `action` is correct in this case. Could you please revert this change?

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Interval.kt [hunk 2]
```diff
@@ -7,9 +7,11 @@ import com.badoo.reaktive.utils.atomic.AtomicLong
  * Returns an [Observable] that emits `0L` after [startDelayMillis] and
  * ever increasing numbers after each period of time specified by [periodMillis], on a specified [Scheduler].
  *
+ * Default start delay is equal to the specified period
+ *
  * Please refer to the corresponding RxJava [document](http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#interval-long-long-java.util.concurrent.TimeUnit-io.reactivex.Scheduler-).
  */
-fun observableInterval(startDelayMillis: Long, periodMillis: Long, scheduler: Scheduler): Observable<Long> =
+fun observableInterval(periodMillis: Long, startDelayMillis: Long = periodMillis,  scheduler: Scheduler): Observable<Long> =
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Interval.kt:14]: Double space?

File: reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/DoOnAfterCompleteTest.kt [hunk 3]
```diff
@@ -17,7 +17,7 @@ class DoOnAfterCompleteTest : MaybeToMaybeTests by MaybeToMaybeTestsImpl({ doOnA
 
     @Test
     fun does_not_call_action_WHEN_upstream_succeeded() {
-        val isCalled = AtomicBoolean()
+        val isCalled = AtomicBoolean(false)
```
Comment (arkivanov) [inline reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/DoOnAfterCompleteTest.kt:20]: It's `false` by default. Why is this change?

**Review by arkivanov — CHANGES_REQUESTED** [review]

**Comment by arkivanov** [comment]
> IDE shows me unused function warning - should we suppress it?

No need to suppress, it's fine.

**Comment by arkivanov** [comment]
Also could you please run apiDump Gradle task and commit the updated api files? Otherwise build will fail.

File: reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/DoOnAfterCompleteTest.kt (hunk 3 above)
Comment (emartynov) [reply reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/DoOnAfterCompleteTest.kt:20]: I see IDE warning/error (I'm on the EAP). Let me revert such changes.

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/maybe/DoOnAfter.kt (hunk 1 above)
Comment (emartynov) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/maybe/DoOnAfter.kt:86]: Done

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Interval.kt (hunk 2 above)
Comment (emartynov) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Interval.kt:14]: Fixed

File: reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/DoOnAfterCompleteTest.kt (hunk 3 above)
Comment (emartynov) [reply reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/DoOnAfterCompleteTest.kt:20]: Done. Here is error:

Screenshot 2021-10-31 at 17 17 56

**Comment by emartynov** [comment]
@arkivanov all comments are processed. Whenever you have time please check.

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thanks for your contribution!
